### ISSUE TITLE
Pretty Printing a channel causes a Segmentation Fault

### ISSUE THREAD

**Issue by elbow-jason** [manastech/crystal#1126]
```crystal

class Message
  getter origin
  getter command
  getter procedure
  def initialize(@origin, @command, @procedure)
  end
end

puts "starting..."
agent_chan = Channel(Message).new
self_chan  = Channel(Int32).new
pp agent_chan
pp self_chan

increment_one = Message.new(self_chan, :update, ->(x : Int32){ x + 1 })
stop = Message.new(self_chan, :stop,   ->(){})

pp increment_one
pp stop

```

This code causes a segfault. Commenting out the `pp agent_chan` prevents the segfault, but I cannot determine why.

**Comment by asterite** [comment]
Found it! The problem is that UnbufferedChannel does something unsafe: it has a `@value` instance variable that might not be initialized, but we tell the compiler to trust us, with `@value :: T`. This is to avoid a nilable type that might cause it to work a bit slower.

Well, that's the price we pay for using unsafe features.

Two solutions:
1. Don't use the unsafe feature: but let's profile it to see if the performance doesn't drop down.
2. Override `inspect` for UnbufferedChannel.

For now I chose to do 2 because it's the easiest thing to do without dropping performance, basically forwarding `inspect` to `to_s`. Plus I'm not sure what `inspect` should do. I tried to print a channel in Go and it only shows its address, so for now we can live with that :-)

@elbow-jason Thanks for the bug report!
